File: backend/Meetings.Models/Resources/UserAnnouncement.cs
using Meetings.Models.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Meetings.Models.Resources
{
    public class UserAnnouncement
    {
        public Guid UserId { get; set; }
        public Guid AnnouncementId { get; set; }
        public string Description { get; set; }
        public UserDTO User { get; set; }
        public int UserAge { get; set; }
        public int? DistanceFromCurrentUser { get; set; }
    }
}

File: frontend-react/src/models/annoucement/user-announcement.ts
import { Announcement } from 'src/models/annoucement/announcement';
import { User } from 'src/models/user';

export type UserAnnouncement = {
    userId: User['id'];
    announcementId: Announcement['id'];
    description: Announcement['description'];
    userBirthDate: string;
    userFirstName: string;
    userProfileImageSrc: string | null;
};
